### DISCUSSION TITLE
Map a http post to a button (Android TV)

### DISCUSSION THREAD

**Discussion by AleXSR700** [Waboodoo/HTTP-Shortcuts#427] [Q&A] [answered]
Hi everyone,
I was wondering if anybody knows of a way to use one of the shortcuts in the app as an action for a button mapper like flar2's Button Mapper or tvQuickActions?

I have tried but nothing happens when I try to do so on my Android TV.

Or maybe someone knows how to assign a shortcut to a hardware button on the TV's remote using a different app or via adb?

Thanks for your ideas
Alex

**Comment by Waboodoo** [comment]
Hi,
I'm not familiar with either of those apps, and I don't have an Android TV to try things out, but on my phone I got Button Mapper to start a shortcut:
1. In Button Mapper, select "Shortcut" as the type of action
2. Scroll down to the HTTP Shortcuts app and click it
3. In the HTTP Shortcuts app, select the shortcut you want to trigger
4. In the dialog that pops up, select "Legacy", not "Default"

I don't know if this would work on a TV, though. Beyond that, I don't have any more ideas.

**Reply by AleXSR700** [reply]
Thank you for your response. 

I had tried exactly that but Android TV does not seem to support shortcuts properly. So since the shortcut is not created, Button Mapper cannot call it (or so it seems).

**Reply by Waboodoo** [reply, marked as answer]
Does calling other apps (not shortcuts) work? If you just need a single shortcut, you could try enabling the "Allow triggering via secondary launcher app" option, found in the shortcut editor under "Trigger & Execution Settings". This will essentially add another app icon called "Trigger shortcut" to your app launcher. Perhaps you can then call that app from Button Mapper.

**Reply by AleXSR700** [reply]
That works! Awesome!
I apologize for not having seen that option earlier. 

Thank you *very* much :)
